db/db_impl.cc

```diff
@@ -2520,6 +2543,9 @@ Status DBImpl::IngestExternalFile(
 
     // Stop writes to the DB
     WriteThread::Writer w;
+    mutex_.Unlock();
+    mem_mutex_.Lock();
```

Comment:
Yeah me too ;) In the past two days I have been fixing deadlocks in unit tests. Let me think further to see how we can simplify this.